sqstaskmaster/local.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import time
+from collections import defaultdict
+import queue
+
+logger = logging.getLogger(__name__)
+"""
+Simple in memory unbounded SimpleQueue based implementation of AWS SQS for local development and integration testing.
+Not for use with threads or multiprocessing. MessageHandler class timeouts should behave as expected.
+No retry behavior is implemented.
+These classes are for synchronous integration testing of message producer and message consumer interface.
+These classes are not for use in production or for validating operational behavior of producer and consumer system.
+
+TODO Consider adding thread support by calling task_done in LocalMessage.delete
+"""
+
+
+class LocalMessage:
+    """
+    Partial implementation of the AWS SQS Message interface for local testing of the message producer and consumer.
+    This does not implement message behavior beyond the body and attributes used for encoded content.
+
+    Methods used by the MessageHandler class (change_visibility and delete) are No-ops!
```

Comment:
no-op because they trigger the logger but otherwise don't do anything?

Reply:
Yes - the behavior that they provide in SQS is not implemented for the local queue.